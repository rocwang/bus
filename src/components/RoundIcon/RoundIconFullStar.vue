<template>
  <path
    fill="currentColor"
    d="M23.39 11.044L19.73 18.8l-8.193 1.248c-1.469.223-2.057 2.115-.992 3.2l5.927 6.034-1.402 8.524c-.252 1.54 1.301 2.695 2.602 1.974L25 35.755l7.329 4.025c1.3.715 2.854-.433 2.602-1.974l-1.402-8.524 5.927-6.035c1.065-1.084.477-2.976-.992-3.199L30.27 18.8l-3.662-7.756a1.763 1.763 0 0 0-3.218 0z"
  />
</template>

<script>
export default {
  name: "RoundIconFullStar",
};
</script>
