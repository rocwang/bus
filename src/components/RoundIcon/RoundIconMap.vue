<template>
  <path
    fill="currentColor"
    d="M25 12a6.562 6.562 0 0 0-6.563 6.563c0 2.93 4.29 8.27 5.933 10.209a.821.821 0 0 0 1.26 0c1.643-1.939 5.933-7.28 5.933-10.21A6.562 6.562 0 0 0 25 12zm0 8.75a2.187 2.187 0 1 1 0-4.374 2.187 2.187 0 0 1 0 4.374zm-13.952 2.497A1.667 1.667 0 0 0 10 24.795v13.037c0 .59.595.993 1.143.774l7.19-3.273v-12.14a15.773 15.773 0 0 1-1.106-2.417l-6.18 2.471zM25 30.733a2.489 2.489 0 0 1-1.902-.884A55.015 55.015 0 0 1 20 25.854v9.479l10 3.333V25.854a54.982 54.982 0 0 1-3.098 3.996 2.49 2.49 0 0 1-1.902.883zm13.857-10.34l-7.19 3.274v15l7.285-2.914A1.666 1.666 0 0 0 40 34.205V21.168a.833.833 0 0 0-1.143-.774z"
  />
</template>

<script>
export default {
  name: "RoundIconMap",
};
</script>
