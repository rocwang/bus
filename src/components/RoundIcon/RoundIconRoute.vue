<template>
  <path
    fill="currentColor"
    d="M34.375 28.75H28.75a1.88 1.88 0 0 1-1.875-1.875A1.88 1.88 0 0 1 28.75 25h5.625S40 18.73 40 15.625A5.626 5.626 0 0 0 34.375 10a5.626 5.626 0 0 0-5.625 5.625c0 1.494 1.3 3.715 2.654 5.625H28.75a5.634 5.634 0 0 0-5.625 5.625c0 3.1 2.525 5.625 5.625 5.625h5.625a1.88 1.88 0 0 1 1.875 1.875 1.88 1.88 0 0 1-1.875 1.875H20.869A40.453 40.453 0 0 1 18.098 40h16.277c3.1 0 5.625-2.525 5.625-5.625s-2.525-5.625-5.625-5.625zm0-15c1.037 0 1.875.838 1.875 1.875a1.873 1.873 0 0 1-1.875 1.875 1.873 1.873 0 0 1-1.875-1.875c0-1.037.838-1.875 1.875-1.875zM15.625 25A5.626 5.626 0 0 0 10 30.625C10 33.73 15.625 40 15.625 40s5.625-6.27 5.625-9.375A5.626 5.626 0 0 0 15.625 25zm0 7.5a1.873 1.873 0 0 1-1.875-1.875c0-1.037.838-1.875 1.875-1.875s1.875.838 1.875 1.875a1.873 1.873 0 0 1-1.875 1.875z"
  />
</template>

<script>
export default {
  name: "RoundIconRoute"
};
</script>
