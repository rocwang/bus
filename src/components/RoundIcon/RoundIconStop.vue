<template>
  <path
    fill="currentColor"
    d="M39.063 13.75H17.5v-2.813a.94.94 0 0 0-.938-.937h-1.875a.94.94 0 0 0-.937.938v2.812h-2.813a.94.94 0 0 0-.937.938v1.874a.94.94 0 0 0 .938.938h2.812v21.563a.94.94 0 0 0 .938.937h1.874a.94.94 0 0 0 .938-.938V17.5h21.563a.94.94 0 0 0 .937-.938v-1.875a.94.94 0 0 0-.938-.937zM19.375 32.5h18.75V19.375h-18.75V32.5z"
  />
</template>

<script>
export default {
  name: "RoundIconStop"
};
</script>
