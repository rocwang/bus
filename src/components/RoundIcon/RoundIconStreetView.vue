<template>
  <path
    fill="currentColor"
    d="M23.008 21.37c-.198.227-.42.433-.683.586v.986c2.858.4 4.813 1.205 4.813 2.133 0 1.329-4.002 2.406-8.938 2.406-4.936 0-8.938-1.077-8.938-2.406 0-.928 1.955-1.732 4.813-2.133v-.986a2.747 2.747 0 0 1-.683-.587C9.73 22.04 7.2 23.444 7.2 25.075c0 2.278 4.925 4.125 11 4.125s11-1.847 11-4.125c0-1.63-2.53-3.036-6.192-3.706zM18.2 12.7a2.75 2.75 0 1 0 0-5.5 2.75 2.75 0 0 0 0 5.5zm-2.75 8.25v4.125c0 .76.616 1.375 1.375 1.375h2.75c.76 0 1.375-.616 1.375-1.375V20.95c.76 0 1.375-.616 1.375-1.375V15.45c0-1.14-.923-2.062-2.063-2.062h-.507c-.475.216-.999.343-1.555.343-.556 0-1.08-.127-1.555-.343h-.507c-1.14 0-2.063.923-2.063 2.062v4.125c0 .76.616 1.375 1.375 1.375z"
  />
</template>

<script>
export default {
  name: "RoundIconStreetView",
};
</script>
