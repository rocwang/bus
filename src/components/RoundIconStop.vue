<template>
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 50 50">
    <path
      fill="#FFDB1B"
      d="M25 0a25 25 0 1 0 25 24.984C49.991 11.184 38.801 0 25 0"
    />
    <path
      fill="#001120"
      d="M39.063 13.75H17.5v-2.813a.94.94 0 0 0-.938-.937h-1.875a.94.94 0 0 0-.937.938v2.812h-2.813a.94.94 0 0 0-.937.938v1.874a.94.94 0 0 0 .938.938h2.812v21.563a.94.94 0 0 0 .938.937h1.874a.94.94 0 0 0 .938-.938V17.5h21.563a.94.94 0 0 0 .937-.938v-1.875a.94.94 0 0 0-.938-.937zM19.375 32.5h18.75V19.375h-18.75V32.5z"
    />
  </svg>
</template>

<script>
export default {
  name: "RoundIconStop"
};
</script>
