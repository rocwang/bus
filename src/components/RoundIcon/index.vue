<template>
  <svg
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 50 50"
    :class="$style[direction]"
  >
    <path
      :fill="bg"
      d="M25 0a25 25 0 1 0 25 24.984C49.991 11.184 38.801 0 25 0"
    />
    <component :is="pathComponent" :style="{ color }" />
  </svg>
</template>

<script>
import RoundIconFullStar from "./RoundIconFullStar";
import RoundIconMap from "./RoundIconMap";
import RoundIconRoute from "./RoundIconRoute";
import RoundIconStop from "./RoundIconStop";
import RoundIconStreetView from "./RoundIconStreetView";

export default {
  name: "RoundIcon",
  components: {
    RoundIconFullStar,
    RoundIconMap,
    RoundIconRoute,
    RoundIconStop,
    RoundIconStreetView
  },
  props: {
    name: {
      type: String,
      required: true,
      validator: function(name) {
        return ["FullStar", "Map", "Route", "Stop", "StreetView"].includes(
          name
        );
      }
    },
    direction: {
      type: String,
      default: "up",
      validator(value) {
        return ["up", "down", "left", "right"].includes(value);
      }
    },
    bg: {
      type: String,
      default: "#000"
    },
    color: {
      type: String,
      default: "#fff"
    }
  },
  computed: {
    pathComponent() {
      return `RoundIcon${this.name}`;
    }
  }
};
</script>

<style module>
.right {
  transform: rotate(90deg);
}

.down {
  transform: rotate(180deg);
}

.left {
  transform: rotate(-90deg);
}
</style>
