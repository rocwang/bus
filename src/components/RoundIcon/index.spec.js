import { mount } from "@vue/test-utils";
import RoundIcon from "./index";
import RoundIconFullStar from "./RoundIconFullStar";
import RoundIconMap from "./RoundIconMap";
import RoundIconRoute from "./RoundIconRoute";
import RoundIconStop from "./RoundIconStop";
import RoundIconStreetView from "./RoundIconStreetView";

describe("Icon.vue", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(RoundIcon, {
      propsData: {
        bg: "green",
        color: "red",
        direction: "down",
        name: "Map"
      }
    });
  });

  it("matches the snapshot", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("has a background", () => {
    expect(wrapper.find("path").attributes("fill")).toBe("green");
  });

  it("has a color", () => {
    expect(
      wrapper
        .findAll("path")
        .at(1)
        .attributes("style")
    ).toMatch(/color:.*/);
  });

  it("has a direction", () => {
    expect(wrapper.classes()).toContain(wrapper.vm.$style.down);
  });

  it("has path components that match the snapshots", () => {
    [
      RoundIconFullStar,
      RoundIconMap,
      RoundIconRoute,
      RoundIconStop,
      RoundIconStreetView
    ].forEach(pathComponents => {
      expect(mount(pathComponents).element).toMatchSnapshot(
        pathComponents.name
      );
    });
  });
});
