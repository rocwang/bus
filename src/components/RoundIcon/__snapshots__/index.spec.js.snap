// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon.vue has path components that match the snapshots: RoundIconFullStar 1`] = `
<path
  d="M23.39 11.044L19.73 18.8l-8.193 1.248c-1.469.223-2.057 2.115-.992 3.2l5.927 6.034-1.402 8.524c-.252 1.54 1.301 2.695 2.602 1.974L25 35.755l7.329 4.025c1.3.715 2.854-.433 2.602-1.974l-1.402-8.524 5.927-6.035c1.065-1.084.477-2.976-.992-3.199L30.27 18.8l-3.662-7.756a1.763 1.763 0 0 0-3.218 0z"
  fill="currentColor"
/>
`;

exports[`Icon.vue has path components that match the snapshots: RoundIconMap 1`] = `
<path
  d="M25 12a6.562 6.562 0 0 0-6.563 6.563c0 2.93 4.29 8.27 5.933 10.209a.821.821 0 0 0 1.26 0c1.643-1.939 5.933-7.28 5.933-10.21A6.562 6.562 0 0 0 25 12zm0 8.75a2.187 2.187 0 1 1 0-4.374 2.187 2.187 0 0 1 0 4.374zm-13.952 2.497A1.667 1.667 0 0 0 10 24.795v13.037c0 .59.595.993 1.143.774l7.19-3.273v-12.14a15.773 15.773 0 0 1-1.106-2.417l-6.18 2.471zM25 30.733a2.489 2.489 0 0 1-1.902-.884A55.015 55.015 0 0 1 20 25.854v9.479l10 3.333V25.854a54.982 54.982 0 0 1-3.098 3.996 2.49 2.49 0 0 1-1.902.883zm13.857-10.34l-7.19 3.274v15l7.285-2.914A1.666 1.666 0 0 0 40 34.205V21.168a.833.833 0 0 0-1.143-.774z"
  fill="currentColor"
/>
`;

exports[`Icon.vue has path components that match the snapshots: RoundIconRoute 1`] = `
<path
  d="M34.375 28.75H28.75a1.88 1.88 0 0 1-1.875-1.875A1.88 1.88 0 0 1 28.75 25h5.625S40 18.73 40 15.625A5.626 5.626 0 0 0 34.375 10a5.626 5.626 0 0 0-5.625 5.625c0 1.494 1.3 3.715 2.654 5.625H28.75a5.634 5.634 0 0 0-5.625 5.625c0 3.1 2.525 5.625 5.625 5.625h5.625a1.88 1.88 0 0 1 1.875 1.875 1.88 1.88 0 0 1-1.875 1.875H20.869A40.453 40.453 0 0 1 18.098 40h16.277c3.1 0 5.625-2.525 5.625-5.625s-2.525-5.625-5.625-5.625zm0-15c1.037 0 1.875.838 1.875 1.875a1.873 1.873 0 0 1-1.875 1.875 1.873 1.873 0 0 1-1.875-1.875c0-1.037.838-1.875 1.875-1.875zM15.625 25A5.626 5.626 0 0 0 10 30.625C10 33.73 15.625 40 15.625 40s5.625-6.27 5.625-9.375A5.626 5.626 0 0 0 15.625 25zm0 7.5a1.873 1.873 0 0 1-1.875-1.875c0-1.037.838-1.875 1.875-1.875s1.875.838 1.875 1.875a1.873 1.873 0 0 1-1.875 1.875z"
  fill="currentColor"
/>
`;

exports[`Icon.vue has path components that match the snapshots: RoundIconStop 1`] = `
<path
  d="M39.063 13.75H17.5v-2.813a.94.94 0 0 0-.938-.937h-1.875a.94.94 0 0 0-.937.938v2.812h-2.813a.94.94 0 0 0-.937.938v1.874a.94.94 0 0 0 .938.938h2.812v21.563a.94.94 0 0 0 .938.937h1.874a.94.94 0 0 0 .938-.938V17.5h21.563a.94.94 0 0 0 .937-.938v-1.875a.94.94 0 0 0-.938-.937zM19.375 32.5h18.75V19.375h-18.75V32.5z"
  fill="currentColor"
/>
`;

exports[`Icon.vue has path components that match the snapshots: RoundIconStreetView 1`] = `
<path
  d="M23.008 21.37c-.198.227-.42.433-.683.586v.986c2.858.4 4.813 1.205 4.813 2.133 0 1.329-4.002 2.406-8.938 2.406-4.936 0-8.938-1.077-8.938-2.406 0-.928 1.955-1.732 4.813-2.133v-.986a2.747 2.747 0 0 1-.683-.587C9.73 22.04 7.2 23.444 7.2 25.075c0 2.278 4.925 4.125 11 4.125s11-1.847 11-4.125c0-1.63-2.53-3.036-6.192-3.706zM18.2 12.7a2.75 2.75 0 1 0 0-5.5 2.75 2.75 0 0 0 0 5.5zm-2.75 8.25v4.125c0 .76.616 1.375 1.375 1.375h2.75c.76 0 1.375-.616 1.375-1.375V20.95c.76 0 1.375-.616 1.375-1.375V15.45c0-1.14-.923-2.062-2.063-2.062h-.507c-.475.216-.999.343-1.555.343-.556 0-1.08-.127-1.555-.343h-.507c-1.14 0-2.063.923-2.063 2.062v4.125c0 .76.616 1.375 1.375 1.375z"
  fill="currentColor"
/>
`;

exports[`Icon.vue matches the snapshot 1`] = `
<svg
  class="down"
  viewBox="0 0 50 50"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M25 0a25 25 0 1 0 25 24.984C49.991 11.184 38.801 0 25 0"
    fill="green"
  />
   
  <path
    d="M25 12a6.562 6.562 0 0 0-6.563 6.563c0 2.93 4.29 8.27 5.933 10.209a.821.821 0 0 0 1.26 0c1.643-1.939 5.933-7.28 5.933-10.21A6.562 6.562 0 0 0 25 12zm0 8.75a2.187 2.187 0 1 1 0-4.374 2.187 2.187 0 0 1 0 4.374zm-13.952 2.497A1.667 1.667 0 0 0 10 24.795v13.037c0 .59.595.993 1.143.774l7.19-3.273v-12.14a15.773 15.773 0 0 1-1.106-2.417l-6.18 2.471zM25 30.733a2.489 2.489 0 0 1-1.902-.884A55.015 55.015 0 0 1 20 25.854v9.479l10 3.333V25.854a54.982 54.982 0 0 1-3.098 3.996 2.49 2.49 0 0 1-1.902.883zm13.857-10.34l-7.19 3.274v15l7.285-2.914A1.666 1.666 0 0 0 40 34.205V21.168a.833.833 0 0 0-1.143-.774z"
    fill="currentColor"
    style="color: red;"
  />
</svg>
`;
